File: java/src/test/java/sammancoaching/PersonTest.java
package sammancoaching;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class PersonTest {
    String swedishPersonalNumber = "19511111-7668";
    int role = Person.USER_ROLE_MANAGER;
    String phoneNumber = "00467123456";
    @Test
    void birthYear() {
        Person person = new Person(role, swedishPersonalNumber, phoneNumber);
        assertEquals(1951, person.birthYear());
    }
    @Test
    void countryCode() {
        Person person = new Person(role, swedishPersonalNumber, phoneNumber);

        assertEquals("+46", person.countryCode());
    }

    @Test
    void managersCanDeleteUsers() {
        Person person = new Person(role, swedishPersonalNumber, phoneNumber);
        assertEquals(true, person.canDeleteUsers());
    }
    @Test
    void adminCanDeleteUsers() {
        Person person = new Person(Person.USER_ROLE_ADMIN, swedishPersonalNumber, phoneNumber);
        assertEquals(true, person.canDeleteUsers());
    }
    @Test
    void salesCantDeleteUsers() {
        Person person = new Person(Person.USER_ROLE_SALES, swedishPersonalNumber, phoneNumber);
        assertEquals(false, person.canDeleteUsers());
    }
    @Test
    void engineersCantDeleteUsers() {
        Person person = new Person(Person.USER_ROLE_ENGINEER, swedishPersonalNumber, phoneNumber);
        assertEquals(false, person.canDeleteUsers());
    }

    @Test
    void standardizePhoneNumber() {
        Person person = new Person(role, swedishPersonalNumber, "+46123456");
        assertEquals("+46", person.countryCode());
    }
    @Test
    void standardizeLocalPhoneNumber() {
        Person person = new Person(role, swedishPersonalNumber, "0123456");
        assertEquals("", person.countryCode());
    }

}

File: kotlin/src/test/kotlin/PersonTest.kt
import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test
import sammancoaching.Person
import sammancoaching.Person.Role

internal class PersonTest {
    private val swedishPersonalNumber = "19511111-7668"
    private val role = Role.MANAGER
    private val phoneNumber = "00467123456"

    @Test
    fun birthYear() {
        val person = Person.createFromSwedishPersonalNumber(role, swedishPersonalNumber, phoneNumber)
        assertEquals(1951, person.birthYear())
    }

    @Test
    fun countryCode() {
        val person = Person.createFromSwedishPersonalNumber(role, swedishPersonalNumber, phoneNumber)
        assertEquals("+46", person.countryCode())
    }

    @Test
    fun managersCanDeleteUsers() {
        val person = Person.createFromSwedishPersonalNumber(role, swedishPersonalNumber, phoneNumber)
        assertTrue(person.canDeleteUsers())
    }

    @Test
    fun adminCanDeleteUsers() {
        val person = Person.createFromSwedishPersonalNumber(Role.ADMIN, swedishPersonalNumber, phoneNumber)
        assertTrue(person.canDeleteUsers())
    }

    @Test
    fun salesCantDeleteUsers() {
        val person = Person.createFromSwedishPersonalNumber(Role.SALES, swedishPersonalNumber, phoneNumber)
        assertFalse(person.canDeleteUsers())
    }

    @Test
    fun engineersCantDeleteUsers() {
        val person = Person.createFromSwedishPersonalNumber(Role.ENGINEER, swedishPersonalNumber, phoneNumber)
        assertFalse(person.canDeleteUsers())
    }

    @Test
    fun standardizePhoneNumber() {
        val person = Person.createFromSwedishPersonalNumber(role, swedishPersonalNumber, "+46123456")
        assertEquals("+46", person.countryCode())
    }

    @Test
    fun standardizeLocalPhoneNumber() {
        val person = Person.createFromSwedishPersonalNumber(role, swedishPersonalNumber, "0123456")
        assertEquals("", person.countryCode())
    }
}
